src/Pact/Analyze/Check.hs

```diff
@@ -415,24 +415,33 @@ moduleFunChecks tables modTys propDefs = for modTys $
   -- @checkFunction@. We need to more carefully enforce this is true!
   argTys' <- for argTys $ \(Pact.Arg name ty _info) ->
     case maybeTranslateType ty of
-      Just ety -> pure (name, ety)
+      Just ety -> pure (Unmunged name, ety)
       Nothing  -> throwError $
         TypeTranslationFailure "couldn't translate argument type" ty
 
-  resultTy' <- case maybeTranslateType resultTy of
-    Just ety -> pure ("result", 0, ety)
+  resultBinding <- case maybeTranslateType resultTy of
+    Just ety -> pure $ Binding 0 (Unmunged "result") (Munged "result") ety
     Nothing  -> throwError $
       TypeTranslationFailure "couldn't translate result type" resultTy
 
-  let env :: [(Text, VarId, EType)]
-      env = resultTy' :
-        ((\(vid, (text, ty)) -> (text, vid, ty)) <$> zip vids argTys')
+  -- NOTE: At the moment, we leave all variables except for the toplevel args
+  -- under analysis as the original munged/SSA'd variable names. And result,
+  -- which we introduce. We also rely on this assumption in Translate's
+  -- mkTranslateEnv.
 
-      nameEnv :: Map Text VarId
-      nameEnv = Map.fromList $ fmap (\(name, vid, _) -> (name, vid)) env
+  let env :: [Binding]
+      env = resultBinding :
+        ((\(vid, (Unmunged nm, ty)) -> Binding vid (Unmunged nm) (Munged nm) ty) <$>
+          zip vids argTys')
 
-      idEnv :: Map VarId EType
-      idEnv = Map.fromList $ fmap (\(_, vid, ty) -> (vid, ty)) env
+      --
+      -- TODO: should the map sent to parsing should be Un/Munged, instead of Text?
```

Comment:
I think we should pass the `Unmunged` name since the `Text` here is what the user would type.

Reply:
Yeah it would be `Unmunged`. Should `Unmunged` extend into that part of the program (vs continuing to use `Text` there)? I suppose so?

Reply:
Yeah, I think it would

Reply:
I see what you're saying now. Will address. Thanks